### ISSUE TITLE
Unable to define custom colors while using defaults

### ISSUE THREAD

**Issue by kencx** [rebelot/kanagawa.nvim#8]
Hi, I'm trying to change the default background color, while also using some of the original defaults to customize the existing hl-groups. Here's a minimal config to show what I mean:

```lua
local default_colors = require("kanagawa.colors")

local colors = {
	sumiInk1 = "#272727", -- new background color
}

local overrides = {
	Search = { bg = default_colors.surimiOrange }, -- customize search highlight color
}

require("kanagawa").setup({
	overrides = overrides,
	colors = colors,
})

vim.cmd("colorscheme kanagawa")
```

The customizations in `overrides` are applied, but the new background color in `colors` is not applied. However, this issue is fixed when I define my own colors as such, without reference to `default_colors`:

```lua
local colors = {
	sumiInk1 = "#272727", 
	surimiOrange = "#FFA066",
}

local overrides = {
	Search = { bg = colors.surimiOrange },
}
```

It seems that reference to `default_colors` is overriding my customizations in `colors`. Is there another way to do this that I'm missing?

Thank you for creating this colorscheme! I'm really loving it.

**Comment by rebelot** [comment]
Requiring `'kanagawa.colors'` runs the module _before_ the config table, thus default colors are assigned to "semantic" colors like fg, bg, visual, etc. I need to make some modifications in the internal logic to defer color assignment.

For the moment, if you want to declare `default_colors`, you can modify the bg color directly like this:
```lua
local default_colors = require("kanagawa.colors")

default_colors.bg = "#272727" -- new background color

local overrides = {
	Search = { bg = default_colors.surimiOrange }, -- customize search highlight color
}

require("kanagawa").setup({
	overrides = overrides
})

vim.cmd("colorscheme kanagawa")
```

**Comment by rebelot** [comment]
Fixed in d1d785c0a71f28f8f535c090f6bb22618f69dd4f

please note that it is now best to use `require('kanagawa').colors` instead of `require('kanagawa.colors')`.

**Comment by kencx** [comment]
Thanks for the quick fix, it works great!
